lib/index.js

```diff
@@ -253,13 +244,18 @@ class ElectronDownloader {
   }
 
   moveFileToCache (filename, target, cb, onSuccess) {
-    debug('moving', filename, 'from', this.tmpdir, 'to', target)
-    fs.unlink(target, (err) => {
-      if (err != null && err.code !== 'ENOENT') return cb(err)
-      fs.move(path.join(this.tmpdir, filename), target, (err) => {
-        if (err) return cb(err)
-        onSuccess(cb)
-      })
+    const cache = this.cache
+    debug('moving', filename, 'from', cache, 'to', target)
+    fs.rename(path.join(cache, filename), target, (err) => {
+      if (err) {
+        try {
+          fs.unlinkSync(cache)
+        } catch (error) {
+          console.error(`Error deleting cache file: ${error.message}`)
```

Comment:
This needs to show the filename that couldn't be removed so the user can deal with it themselves.